roles/update-pkgs/tasks/main.yml

```diff
@@ -0,0 +1,27 @@
+- name: Update packages and kernel to latest available versions
+  package:
+    name: '*'
+    state: latest
+  when: ansible_pkg_mgr in ['yum', 'dnf']
+
+- name: Check if reboot required
+  shell: needs-restarting -r
+  register: reboot_check
+  ignore_errors: yes
+  when: ansible_distribution in ['CentOS', 'RHEL']
+
+- name: Reboot if necessary
+  reboot:
+    reboot_timeout: 600
+    test_command: ping -c 4 {{ ansible_default_ipv4.address }}
+  when: reboot_check.rc == 1 and ansible_distribution in ['CentOS', 'RHEL']
+
+- name: Wait for the system to come back online after reboot
+  wait_for:
```

Comment:
if previous module itself reboots and waits, then why we need this additional one?

Reply:
This was mostly an additional check on SSHd  in particular, I realise it's mostly redundant because the script would anyway use the same. 
Thanks!